### PR TITLE
Helper tooltips

### PR DESCRIPTION
**Pull request by moreofmorris**
### Purpose
This introduces new Helper tooltips for metrics in Analyze. I'm trying to make the tooltips match up nicely with the ones that are in the graphs and also have that "lively" feel with some added movement to them. The helper tooltips also recognise if they're being clipped by the outer chart card and so then will right align to still be seen.

### Notes
To overcome the problem of the tooltips running off the each of the page if they're too far to the right of the interface I had to put some logic in that detects the chart card and then right align tooltip. I tried every which way to make this as clean as possible but had to resort to going up the DOM until it found an `<article>` (which every chart card is) and using that as the boundary. Also, because the metric object that is passed is different depending on the context in the app I couldn't rely on the `metric.key` (eg. `engagement_rate`) but could rely on the `metric.label` as that was always available, that also allowed for subtle helper text differences so actually works out well. Ideally we'd have some sort of central, formalised place for all metric related data to draw from (and the helper text would be part of that) but we don't have that today.

The text for these can be found here https://paper.dropbox.com/doc/Analyze-Helpers-1.0--AelhXQPsLPkIUUQmwC_8PjYaAg-aeXyAOGdzW8YGKyDcJR8d

### Staging Deployment
https://task-helper-tooltips-analyze.dev.buffer.com/ 😄

### CODE REVIEW

File: packages/shared-components/Helper/index.jsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    if (tooltipEl) {
+      const tooltipRect = tooltipEl.getBoundingClientRect();
+      return tooltipRect.width;
+    }
+    return false;
+  }
+
+  isTooltipClipping = () => {
+    // this is required to find boundary so to left or right align tooltip to avoid cutoff
+    const tooltipEl = ReactDOM.findDOMNode(this.tooltipEl);
```
Comment (federicoweber) [inline packages/shared-components/Helper/index.jsx]: usually `findDOMNode`, is a bit discouraged in React, but I can't really think on a different way to do this 🤔.

File: packages/shared-components/Helper/helpers.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/* eslint quote-props: ["error", "consistent"] */
```
[the remark this answers is not in the document]
Comment (moreofmorris) [reply packages/shared-components/Helper/helpers.js]: Happy to do that bro 👍

File: packages/shared-components/Helper/index.jsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  transform: rotate(45deg);
+
+  ${props => props.show && css`
+    display: block;
+    animation: ${animationDropper} 250ms linear;
+    opacity: 1;
+  `}
+`;
+
+function normaliseLabel(label) {
```
[the remark this answers is not in the document]
Comment (moreofmorris) [reply packages/shared-components/Helper/index.jsx]: Yes, haha. I'll update it to American English. It should be in code anyhow 👍Great catch!

File: packages/shared-components/Helper/index.jsx (hunk 1 above)
Comment (moreofmorris) [reply packages/shared-components/Helper/index.jsx]: I know, totally agree on that Feds! I tried to use just the `ref` stuff by itself (in all its various forms) but it wasn't working at all. I'll spend some more time looking for solutions 👍

**Pull request merged**